ui/src/main/java/org/pentaho/di/ui/spoon/delegates/SpoonDBDelegate.java

```diff
@@ -474,12 +475,33 @@ public void saveConnection( DatabaseMeta db, String versionComment ) {
 
   }
 
-  public void newConnection() {
-    HasDatabasesInterface hasDatabasesInterface = spoon.getActiveHasDatabasesInterface();
-    if ( hasDatabasesInterface == null && spoon.rep == null ) {
-      return;
+  public void newConnection( ) {
+    DatabaseMeta databaseMeta = new DatabaseMeta();
+    getDatabaseDialog().setDatabaseMeta( databaseMeta );
+    String con_name = getDatabaseDialog().open();
+    if ( !Utils.isEmpty( con_name ) ) {
+      con_name = con_name.trim();
+      databaseMeta.setName( con_name );
+      databaseMeta.setDisplayName( con_name );
+      databaseMeta = getDatabaseDialog().getDatabaseMeta();
+      try {
+        DatabaseManagementInterface databaseManagementInterface = spoon.getBowl().getManager( DatabaseManagementInterface.class );
+
+        if ( databaseMeta.findDatabase( databaseManagementInterface.getDatabases(), con_name ) == null ) {
+          databaseManagementInterface.addDatabase( databaseMeta );
+          if ( spoon.rep != null ) {
+            saveConnectionToRepository( databaseMeta );
```

Comment:
why are we saving to the repository? I thought the repository was "Local".

Reply:
This is an existing flow. Currently, the new connection gets saved in repository if the user is connected to repository. I kept the flow since we haven't groomed the repository stories

Reply:
I think this is in this story. Repository == local, and syncing to local is precisely what this story is supposed to be removing.

Reply:
oh, that isn't this story, it's the "rip out shared/unshared" story that is already supposed to be dev complete. This would add back behavior that the other story removed.

Reply:
I thought we are not changing/breaking the code that works with the repository i.e leaving it as is. 
That is why I didn't touch it in previous "rip out share/unshare" story and this story

Reply:
The plan for removing share/unshare, plus this story, means there should no longer be any way to create "Local" connections anywhere. This isn't "breaking" repository, it's just moving forward with the feature.